pkg/util/pki/parse.go

```diff
@@ -1,13 +1,46 @@
 package pki
 
 import (
+	"crypto"
 	"crypto/rsa"
 	"crypto/x509"
 	"encoding/pem"
 
 	"github.com/jetstack/cert-manager/pkg/util/errors"
 )
 
+func DecodePrivateKeyBytes(keyBytes []byte) (crypto.PrivateKey, error) {
+	// decode the private key pem
+	block, _ := pem.Decode(keyBytes)
+	if block == nil {
+		return nil, errors.NewInvalidData("error decoding private key PEM block")
+	}
+
+	switch block.Type {
+	case "EC PRIVATE KEY":
+		key, err := x509.ParseECPrivateKey(block.Bytes)
+		if err != nil {
+			return nil, errors.NewInvalidData("error parsing ecdsa private key: %s", err.Error())
+		}
+
+		return key, nil
+	case "RSA PRIVATE KEY":
+		key, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+		if err != nil {
+			return nil, errors.NewInvalidData("error parsing rsa private key: %s", err.Error())
+		}
+
+		err = key.Validate()
```

Comment:
Is there a `key.Validate` equivalent for ecdsa? I see https://golang.org/pkg/crypto/ecdsa/#Verify, but not sure if that is for verifying a certificate against a public key?

Reply:
Unfortunately, there's no equivalent for ecdsa. 
You're right as thats for verifying a certificate against the public key